src/components/PopupWithForm.js

```diff
@@ -18,6 +19,15 @@ export default class PopupWithForm extends Popup {
     return inputValues;
   }
 
+  // Maybe this works???
+  setLoading(loading) {
+    if (loading) {
+      this._submitBtn.textContent = "Saving ...";
+    } else {
+      this._submitBtn.textContent = "Save";
+    }
+  }
+
```

Comment:
Needs correction: (line 37) Сlose modal in the case of successful api response in `then` block in `index.js`, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.